### ISSUE TITLE
[TF2] Evading Autobalance

### ISSUE THREAD

**Issue by ghost** [ValveSoftware/Source-1-Games#2697]
When the 5 second cooldown period for autobalance starts on a Casual MM Server, if you use the retry command to rejoin the server, you will no longer be autobalanced to the opposing team, instead, another player will be picked at random. The server should realize if a player if rejoining the server, and still switch them to the other team regardless, rather than picking another player. Ideally, I believe there should be a matchmaking penalty for this behavior instead however.

**Comment by AhRoon** [comment]
Don't report this pls

**Comment by AhRoon** [comment]
Auto-balance sometimes makes teams fair or unfair when your team is less. But when your team have many players (e.g. 11 players in BLU vs. 9 players in RED), You can be moved to opposing team. Then you will think that the Auto-balance System is unfair. That's why I think to keep evading bug should be maintained until the Valve finds right idea. I just want Valve to find right idea to make the game fair.

**Comment by AhRoon** [comment]
@dynacorpindustries

**Comment by dynacorpindustries** [comment]
A bug is not an intended function of code, and should be fixed, regardless of the circumstance. This bug in particular can damage your team composition if you use it, because it picks another player on your team at random instead of you. So rather than you being autobalanced to the other team, the server may end up autobalancing your top player instead, which could change the game completely, which would probably end up harming you in the long run. While this bug may "help" you in rare circumstances, abusing a bug is exploiting, which is simply not right. If you want a solution to autobalance, you should file a suggestion rather than asking me to retract a bug report so that people may continue exploiting it.

**Comment by henriquecm** [comment]
I just disagree with the MM penalty, there shouldn't be any timed penalty in causal, if there is one it should be losing XP or making the player selected more often for auto balance, so if he does again will continuously increase his chance of being autobalanced until he stops.

**Comment by AhRoon** [comment]
Yes. What about adding bot with left players name in Match without making Autobalance? This could be great. Alright, I'm gonna make another Issues with Suggestions.

**Comment by henriquecm** [comment]
Adding a bot would help, even in competitive bots are added to balance the game.
I think the best course of action would be adding a bot and make available an option to switch to the team with less players at any given time.

Something that usually causes imbalance is spawn camp, maybe could be too much, but making so that we can't take damage from the enemy team inside the spawn would also help, we could only self-damage.

**Comment by Ashetf2** [comment]
Maybe Valve can remove that 5 seconds wait time and just make the balance to occur instantly.

**Comment by vbisbest** [comment]
Auto balance should be based on last player joined.

**Comment by LethalManBoob** [comment]
just fill the empty slots with bots
